File: Source/OldCode/Archive/2005 TVA Code Library/Version/CS/TVA.Core_CS/Net/Ftp/Exception.cs
//*******************************************************************************************************
//  Exception.cs
//  Copyright © 2008 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR 2W-C
//       Phone: 423/751-2827
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  05/22/2003 - James R Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

using System;

namespace TVA.Net.Ftp
{
    public abstract class ExceptionBase : Exception
    {
        private Response m_ftpResponse = null;

        internal ExceptionBase(string message)
            : base(message)
        {
        }

        internal ExceptionBase(string message, ExceptionBase inner)
            : base(message, inner)
        {
        }

        internal ExceptionBase(string message, Response ftpResponse)
            : base(message)
        {
            m_ftpResponse = ftpResponse;
        }

        public string ResponseMessage
        {
            get
            {
                if (!(m_ftpResponse == null))
                    return m_ftpResponse.Message;
                else
                    return "";
            }
        }
    }

    public class InvalidResponseException : ExceptionBase
    {
        internal InvalidResponseException(string message, Response ftpResponse)
            : base(message, ftpResponse)
        {
        }
    }

    public class AuthenticationException : ExceptionBase
    {
        internal AuthenticationException(string message, Response ftpResponse)
            : base(message, ftpResponse)
        {
        }
    }

    public class FileNotFoundException : ExceptionBase
    {
        internal FileNotFoundException(string remoteFile)
            : base("Remote file (" + remoteFile + ") not found.  Try refreshing the directory.")
        {
        }
    }

    public class ServerDownException : ExceptionBase
    {
        internal ServerDownException(Response ftpResponse)
            : this("FTP service was down.", ftpResponse)
        {
        }

        internal ServerDownException(string message, Response ftpResponse)
            : base(message, ftpResponse)
        {
        }
    }

    public class CommandException : ExceptionBase
    {

        internal CommandException(string message, Response ftpResponse)
            : base(message, ftpResponse)
        {
        }
    }

    public class DataTransferException : ExceptionBase
    {
        internal DataTransferException()
            : base("Data transfer error: previous transfer not finished.")
        {
        }

        internal DataTransferException(string message, Response ftpResponse)
            : base(message, ftpResponse)
        {
        }
    }

    public class UserAbortException : ExceptionBase
    {
        internal UserAbortException()
            : base("File Transfer aborted by user.")
        {
        }
    }

    public class ResumeNotSupportedException : ExceptionBase
    {
        internal ResumeNotSupportedException(Response ftpResponse)
            : base("Data transfer error: server does not support resuming.", ftpResponse)
        {
        }
    }
}

File: Source/OldCode/Archive/2005 TVA Code Library/Version/2.0/Local/Source/TVA.Core/Net/Ftp/Exception.vb
' James Ritchie Carroll - 2003
Option Explicit On 

Namespace Net.Ftp

    Public Class Exception

        Inherits System.Exception

        Private m_ftpResponse As Response = Nothing

        Friend Sub New(ByVal message As String)

            MyBase.New(message)

        End Sub

        Friend Sub New(ByVal message As String, ByVal inner As Exception)

            MyBase.New(message, inner)

        End Sub

        Friend Sub New(ByVal message As String, ByVal ftpResponse As Response)

            MyBase.New(message)
            m_ftpResponse = ftpResponse

        End Sub

        Public ReadOnly Property ResponseMessage() As String
            Get
                If Not (m_ftpResponse Is Nothing) Then
                    Return m_ftpResponse.Message
                Else
                    Return ""
                End If
            End Get
        End Property

        Public Overrides ReadOnly Property Message() As String
            Get
                Return MyBase.Message
            End Get
        End Property

    End Class

    Public Class InvalidResponseException

        Inherits Exception

        Friend Sub New(ByVal message As String, ByVal ftpResponse As Response)

            MyBase.New(message, ftpResponse)

        End Sub

    End Class

    Public Class AuthenticationException

        Inherits Exception

        Friend Sub New(ByVal message As String, ByVal ftpResponse As Response)

            MyBase.New(message, ftpResponse)

        End Sub

    End Class

    Public Class FileNotFoundException

        Inherits Exception

        Friend Sub New(ByVal remoteFile As String)

            MyBase.New("Remote file (" & remoteFile & ") not found.  Try refreshing the directory.")

        End Sub

    End Class

    Public Class ServerDownException

        Inherits Exception

        Friend Sub New(ByVal ftpResponse As Response)

            Me.New("FTP service was down.", ftpResponse)

        End Sub

        Friend Sub New(ByVal message As String, ByVal ftpResponse As Response)

            MyBase.New(message, ftpResponse)

        End Sub

    End Class

    Public Class CommandException

        Inherits Exception

        Friend Sub New(ByVal message As String, ByVal ftpResponse As Response)

            MyBase.New(message, ftpResponse)

        End Sub

    End Class

    Public Class DataTransferException

        Inherits Exception

        Friend Sub New()

            MyBase.New("Data transfer error: previous transfer not finished.")

        End Sub

        Friend Sub New(ByVal message As String, ByVal ftpResponse As Response)

            MyBase.New(message, ftpResponse)

        End Sub

    End Class

    Public Class UserAbortException

        Inherits Exception

        Friend Sub New()

            MyBase.New("File Transfer aborted by user.")

        End Sub

    End Class

    Public Class ResumeNotSupportedException

        Inherits Exception

        Friend Sub New(ByVal ftpResponse As Response)

            MyBase.New("Data transfer error: server does not support resuming.", ftpResponse)

        End Sub

    End Class

End Namespace